include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -0,0 +1,224 @@
+    }
+    //!\}
+
+    /*!\name Invocation
+     * \{
+     */
+    /*!\brief Computes the pairwise sequence alignment for the given range over indexed sequence pairs.
+     * \tparam indexed_sequence_pairs_t The type of indexed_sequence_pairs; must model
+     *                                  seqan3::detail::indexed_sequence_pair_range.
+     * \tparam callback_t The type of the callback function that is called with the alignment result; must model
+    *                    std::invocable with seqan3::alignment_result as argument.
+     *
+     * \param[in] indexed_sequence_pairs A range over indexed sequence pairs to be aligned.
+     * \param[in] callback The callback function to be invoked with each computed alignment result.
+     *
+     * \throws std::bad_alloc during allocation of the alignment matrices or
+     *         seqan3::invalid_alignment_configuration if an invalid configuration for the given sequences is detected.
+     *
+     * \details
+     *
+     * Uses the standard dynamic programming algorithm to compute the pairwise sequence alignment for each
+     * sequence pair. The space and runtime complexities depend on the selected configurations (see below).
+     * For every computed alignment the given callback is invoked with the respective alignment result.
+     *
+     * ### Exception
+     *
+     * Strong exception guarantee. Might throw std::bad_alloc or seqan3::invalid_alignment_configuration.
+     *
+     * ### Thread-safety
+     *
+     * Calls to this functions in a concurrent environment are not thread safe. Instead use a copy of the alignment
+     * algorithm type.
+     *
+     * ### Complexity
+     *
+     * The following table lists the runtime and space complexities for the banded and unbanded algorithm dependent
+     * on the configured seqan3::align_cfg::result per sequence pair.
+     * Let `n` be the length of the first sequence, `m` be the length of the second sequence and `k` be the size of
+     * the band.
+     *
+     * |                        | unbanded         | banded            |
+     * |:----------------------:|:----------------:|:-----------------:|
+     * |runtime                 |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     * |space (score only)      |\f$ O(m) \f$      |\f$ O(k) \f$       |
+     * |space (end positions)   |\f$ O(m) \f$      |\f$ O(k) \f$       |
+     * |space (begin positions) |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     * |space (alignment)       |\f$ O(n*m) \f$    |\f$ O(n*k) \f$     |
+     */
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, alignment_result_type>
+    //!\endcond
+    void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
+    {
+        using result_value_t = typename alignment_result_value_type_accessor<alignment_result_type>::type;
+        using std::get;
```

Comment:
Is the sequence pair something else than a std pair/tuples sometimes?

Reply:
I can't know. It is generic code. It could be seqan3::pod_tuple for example or any other third-library tuple implementation. So I have to call get unqualified. However, since get is a function template with an explicit template argument it must be first introduced into the scope otherwise the compiler does not know that he should actually look for a function called get and tries to evaluate the expression `get < 0 > (...)` like `get smaller than 0 greater than (expression)`, which will fail to compile.